src/MajorConstraintUpdater.php

```diff
@@ -0,0 +1,100 @@
+                          updated all the other packages such that there arent conflicts'
+            )
+            ->setDefinition([
+                new InputOption(
+                    'composer-json',
+                    null,
+                    InputOption::VALUE_OPTIONAL,
+                    'Composer json file location',
+                    Factory::getComposerFile()
+                ),
+                new InputOption(
+                    'composer-lock',
+                    null,
+                    InputOption::VALUE_OPTIONAL,
+                    'Composer lock file location',
+                    Factory::getLockFile(Factory::getComposerFile())
+                ),
+                new InputOption(
+                    'constraint',
+                    null,
+                    InputOption::VALUE_IS_ARRAY | InputOption::VALUE_OPTIONAL,
+                    'Package/packages to require with a version constraint, e.g. foo/bar:1.0.0 or foo/bar=1.0.0'
+                ),
+            ])
+            ->setHelp(
+                <<<EOT
+                    The <info>major-update</info> command updates your composer.json file such that other packages are updated
+                    to the newest versions without a need to resolve conflicts manually.  For example, if you’re upgrading package/package from version 9 to 10 and there are conflicts with other packages,
+                    you can use major-update --constraint=php:^8.1 --constraint=package/package:^10.0.0. This command will automatically resolve the conflicts in composer.json, saving you the trouble of doing it manually
+                    EOT
+            );
+    }
+
+    /**
+     * @throws Exception
+     */
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $composerPath = $input->getOption('composer-json');
+        $composerLock = $input->getOption('composer-lock');
+
+        $constraints = [];
+
+        foreach ($input->getOption('constraint') as $option) {
+            $input = explode(':', $option);
+            if(!isset($input[0], $input[1])) {
+                throw new Exception('Wrong input constraint, should be as package:version');
+            }
+            $constraints[$input[0]] = $input[1];
+        }
+        $composerJson = new ComposerJson($composerPath, $composerLock);
+        $output->writeln('Building composer.json for update');
+        file_put_contents(
+            $composerPath,
+            $composerJson->replaceVersions($constraints)
+        );
+
+        $output->writeln('Launching composer update');
+        $this->updateComposer();
+
+        $output->writeln('Rebuilding composer.json from lock');
+        file_put_contents(
+            $composerPath,
+            $composerJson->rebuildFromLock()
+        );
+
+        $output->writeln('Composer.json has been successfully updated!');
+
+        return 1;
+    }
+
+    private function updateComposer(): void
```

Comment:
Added tests, yet they are probably not ideal.
Had an idea to call the UpdateCommand object and run execute, yet for some reason couldn't get this to work and gave up.

Saw that Symfony offers some kind of command tester, yet this would require an external dependency and don't know if it would work either way.